### PR TITLE
Change config navigation to tabs

### PR DESCRIPTION
**Pull request by bramkragten**
image

image

image
image

### CODE REVIEW

**Comment by bramkragten** [comment]
Icon suggestions are welcome

File: src/layouts/hass-tabs-subpage.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+import { Route, HomeAssistant } from "../types";
+import { navigate } from "../common/navigate";
+import "@material/mwc-ripple";
+import { ConfigPageNavigation } from "../panels/config/dashboard/ha-config-navigation";
+import { isComponentLoaded } from "../common/config/is_component_loaded";
+
+@customElement("hass-tabs-subpage")
+class HassTabsSubpage extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property({ type: Boolean }) public showBackButton = true;
```
Comment (balloob) [inline src/layouts/hass-tabs-subpage.ts]: We don't need this one I think.

File: src/layouts/hass-tabs-subpage.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  @property({ type: String, attribute: "back-path" }) public backPath?: string;
+  @property() public backCallback?: () => void;
+  @property({ type: Boolean }) public hassio = false;
+  @property({ type: Boolean }) public showAdvanced = false;
+  @property() public route!: Route;
+  @property() public tabs!: ConfigPageNavigation[];
```
Comment (balloob) [inline src/layouts/hass-tabs-subpage.ts]: Let's rename the type since we can use it on other pages

File: src/layouts/hass-tabs-subpage.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  @property({ type: Boolean, reflect: true }) public narrow = false;
+  @property() private _activeTab: number = -1;
+
+  protected updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (changedProperties.has("route")) {
+      this._activeTab = this.tabs.findIndex((tab) =>
+        this.route.prefix.includes(`/config/${tab.page}`)
```
Comment (balloob) [inline src/layouts/hass-tabs-subpage.ts]: This seems very specific to config page. I think that we should have `tab.page` be the full path.

File: src/layouts/hass-tabs-subpage.ts [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+import { isComponentLoaded } from "../common/config/is_component_loaded";
+
+@customElement("hass-tabs-subpage")
+class HassTabsSubpage extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property({ type: Boolean }) public showBackButton = true;
+  @property({ type: String, attribute: "back-path" }) public backPath?: string;
+  @property() public backCallback?: () => void;
+  @property({ type: Boolean }) public hassio = false;
+  @property({ type: Boolean }) public showAdvanced = false;
```
Comment (balloob) [inline src/layouts/hass-tabs-subpage.ts]: Please add docstrings to each property. It's not clear to me what this property does.

File: src/layouts/hass-tabs-subpage.ts [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        </div>
+      </div>
+      <div class="content">
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  private _tabTapped(ev: MouseEvent): void {
+    navigate(this, `/config/${(ev.currentTarget as any).page}`, true);
```
Comment (balloob) [inline src/layouts/hass-tabs-subpage.ts]: Tab needs absolute path.

File: src/panels/config/ha-panel-config.ts [hunk 6]
```diff
@@ -82,67 +230,45 @@ class HaPanelConfig extends LitElement {
     if (isComponentLoaded(this.hass, "cloud")) {
       this._updateCloudStatus();
     }
+    this.style.setProperty(
```
Comment (balloob) [inline src/panels/config/ha-panel-config.ts]: Why can't we put this in the CSS?

File: src/panels/config/person/ha-config-person.ts [hunk 7]
```diff
@@ -48,9 +51,12 @@ class HaConfigPerson extends LitElement {
...
     return html`
-      <hass-subpage
-        .header=${hass.localize("ui.panel.config.person.caption")}
-        .showBackButton=${!this.isWide}
+      <hass-tabs-subpage
+        .hass=${this.hass}
+        .narrow=${this.narrow}
+        .route=${this.route}
+        back-path="/config"
+        .tabs=${configSections[2]}
```
Comment (balloob) [inline src/panels/config/person/ha-config-person.ts]: Any reason it's a list and not independent exports NAV_INTEGRATIONS etc. List indices are fragile as you might end up changing the entries and now you need to update a lot of places, and typescript can't tell us if we messed up .

File: src/panels/config/person/ha-config-person.ts [hunk 8]
```diff
@@ -231,7 +238,9 @@ ${this.hass!.localize("ui.panel.config.person.confirm_delete2")}`)
         right: 16px;
         z-index: 1;
       }
-
+      ha-fab[narrow] {
+        bottom: 76px;
```
Comment (balloob) [inline src/panels/config/person/ha-config-person.ts]: Which one wins here, `is-wide` bottom or `narrow` below?

Comment (balloob) [reply src/panels/config/person/ha-config-person.ts]: Oh I guess they can never go together, nevermind

**Comment by balloob** [comment]
- [ ] hass-tabs-subpage is having a scrollbar, even if the internal content has their own scrollbar. Debug by going to system prefs -> general -> always show scrollbars
- [ ] Tabs need `cursor: pointer`
- [ ] Config sections are not being shown in "wide" mode with descriptions to the left of the controls
- [ ] Icons in config dashboard need to be `color: var(--secondary-text-color)` and centered in icon slot

**Comment by SeanPM5** [comment]
I think that maybe the icons should maybe be colored with secondary text color (so gray in this case). I feel the icons stand out a tad too much currently, it's taking attention away from the text rather than complementing it. 

For the bottom tabs on mobile, I think it'd be a little nicer if they all had text and not just the active tab. Makes it look more consistent and removes any guesswork on what an icon represents.


**Some icon suggestions and reasonings**

Cloud `mdi:cloud-check-outline` I think checkmark feels more natural than lock, since you are making things available to the cloud.

Integrations `mdi:puzzle` most things seem to use that for add-ons/extensions/integrations. Search "[browser extensions](https://www.google.com/search?tbm=isch&q=browser%20extensions)" or "add-ons" in google images for example.

Areas I would do `mdi:sofa` . I don't think the floorplan icon is easy enough to distinguish in such a small size, when people see a couch they'll think of their living room and it'll be clear what the purpose of areas is.

Automations `mdi:robot` maybe? When most people think of stuff being automated robots come to mind, and there is precedent with [macOS Automator icon](https://www.google.com/search?tbm=isch&q=Automator%20icon). Might be a little too "playful" tho compared to the other icons?

Scenes `mdi:palette` since mostly this will be used for changing colors of lights. Art / paintbrush palette seems fitting? Current choice feels more like an equalizer.

Persons `mdi:account-group` since you are usually adding more than one person.

**Comment by balloob** [comment]
fyi, automation/scene/person icons are based on their domain icons. They need to stay in sync. So it can be changed, but we need to change both.

@SeanPM5 please copy paste the picture of icons besides just mentioning the name, so much easier !  👍

**Comment by SeanPM5** [comment]
| Icon | Name |  MDI icon    |
|--------------|----------------------|-------------|
| cloud    | Cloud         | `mdi:cloud-check-outline` |
| integrations    | Integrations         | `mdi:puzzle` |
| areas             | Areas                | `mdi:sofa` |
| automation             | Automations          | `mdi:robot` |
| scenes             | Scenes               | `mdi:palette` |
| persons              | Persons              | `mdi:account-multiple` |

Some of these like scenes and integrations have "outline" versions available too.

**Comment by balloob** [comment]
I like all your suggestions except for the cloud and the person one. 

For Cloud, the main use case is remote UI and I like to emphasize that it's encrypted and secure.

For persons. Since it's also the domain icon, picking an icon with 2 persons on it might be weird if it's shown for a single person in an entity row

**Comment by balloob** [comment]
Can we ditch the top bar coloring at the following two places and just render the icon with the same background color?

image

Screen Shot 2020-01-28 at 09 55 11

Also, the fab on the mobile screenshot is not equally away from bottom and right edge.

**Comment by iantrich** [comment]
looking great 🎉 nothing constructive to add 😜

File: src/panels/config/ha-panel-config.ts (hunk 6 above)
Comment (bramkragten) [reply src/panels/config/ha-panel-config.ts]: Now it applies to all config pages, and this is a router element

**Review by balloob — APPROVED** [review]

**Comment by SeanPM5** [comment]
> For Cloud, the main use case is remote UI and I like to emphasize that it's encrypted and secure.

This reasoning makes perfect sense, although I do think the meaning of the lock can _potentially_ be misinterpreted, for example being "locked out" of your cloud account (did I pay my bill this month?), or being "locked out" of some features during your free trial period. I suspect those cases will be rare, just something to consider. I like it though and think it's fine to keep the current choice.

> For persons. Since it's also the domain icon, picking an icon with 2 persons on it might be weird if it's shown for a single person in an entity row

Yeah, if it's also the domain icon, it's gonna feel a little bit weird in one place or the other I guess. 🤷‍♂

Current icon choice is more commonly associated with a profile settings page ([image results](https://www.google.com/search?tbm=isch&q=user%20profile%20icon)). On mobile which doesn't have the text underneath the tab icons, I would think that most people are going to assume that goes to their personal profile page. 

Plus the title of the panel is Persons plural, and when you click on it you get a listing of multiple people.. so it feels weird to not have the icon represent that? But as you mentioned, if it gets changed here, it's gonna look weird in an entity row. So I dunno maybe it's best to leave it alone.

**Pull request merged**